epsagon/events/botocore.py

```diff
@@ -1221,6 +1226,39 @@ def update_pool_op(self, args, _):
         self.resource['name'] = request_args['UserPoolId']
         self.resource['metadata'].update(request_args)
 
+    def sign_up_op(self, args, _):
+        """
+        Process User Signup operation
+        :param args: command arguments
+        :param _: unused, kwargs
+        :return: None
+        """
+        _, request_args = args
+        self.resource['name'] = request_args.get('ClientId')
+        self.resource['metadata'].update(request_args)
+
+    def admin_initiate_auth_op(self, args, _):
+        """
+        Process Auth Initiation as Admin
+        :param args: command arguments
+        :param _: unused, kwargs
+        :return: None
+        """
+        _, request_args = args
+        self.resource['name'] = request_args['UserPoolId']
+        self.resource['metadata'].update(request_args)
+
+    def admin_respond_to_auth_challenge_op(self, args, _):
+        """
+        Process Respond to Auth Challenge as Admin
+        :param args: command arguments
+        :param _: unused, kwargs
+        :return: None
+        """
+        _, request_args = args
+        self.resource['name'] = request_args['UserPoolId']
+        self.resource['metadata'].update(request_args)
```

Comment:
Same exact function logic (admin_initiate_auth_op &  admin_respond_to_auth_challenge_op) and both are duplicates of update_pool_op.